types/signed_msg_type.go

```diff
@@ -0,0 +1,22 @@
+package types
+
+type SignedMsgType byte
+
+const (
+	// Types of votes
+	PrevoteType   SignedMsgType = 0x01
+	PrecommitType SignedMsgType = 0x02
+	// To separate (canonicalized) Vote and Proposal messages:
+	ProposalType SignedMsgType = 0x03
```

Comment:
Let's distinguish categories of msg type like "vote-associated", "proposal-associated", "heartbeat-associated". So eg.

```
        // Vote types
	PrevoteType   SignedMsgType = 0x01
	PrecommitType SignedMsgType = 0x02

	// Proposal Types
	ProposalType SignedMsgType = 0x10

        // Heartbeat Types
        HeartbeatType SignedMsgType = 0x20
        ExampleNewHeartbeatType SignedMsgType = 0x21

```